nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -96,58 +101,46 @@ private void initScriptHeader() throws IOException, InvalidFormatException {
 
 	/**
 	 * Attempt to decompress the data from the reader. Supports LZMA algorithm. Will
-	 * eventually support Bzip2 and Zlib.
+	 * eventually support Bzip2 and Zlib. The reader offset has to be set at the
+	 * beginning of the compressed data before calling this function.
 	 * 
+	 * @param offset, the offset at which the compressed data can be found
 	 * @throws IOException
 	 */
-	private byte[] decompressData() throws IOException {
-		this.reader.setPointerIndex(this.headerOffset + NsisScriptHeader.getHeaderSize());
-		byte compressionByte = this.reader.readNextByte();
-		if (NsisConstants.COMPRESSION_LZMA == compressionByte) {
-			long compressedDataOffset;
-			int compressedDataLength;
-			byte[] compressedData;
-			int dictionarySize = this.reader.readNextInt();
-			compressedDataOffset = this.headerOffset + NsisScriptHeader.getHeaderSize()
-					+ NsisConstants.COMPRESSION_LZMA_HEADER_LENGTH;
-			// Flip the MSB to get the length
-			compressedDataLength = (this.scriptHeader.compressedHeaderSize & ~FLAG_IS_COMPRESSED)
-					- NsisConstants.COMPRESSION_LZMA_HEADER_LENGTH;
-			compressedData = this.reader.readByteArray(compressedDataOffset, compressedDataLength);
-			return decompressLZMA(compressedData, compressionByte, dictionarySize);
-		} else if (NsisConstants.COMPRESSION_BZIP2 == compressionByte) {
-			// TODO Bzip2 decompress
-			System.out.println("Decompress Bzip");
-			return new byte[0];
-		} else {// TODO find a was to identify Zlib compressed
-			// TODO Zlib decompress
-			System.out.println("Decompress Zlib");
-			return new byte[0];
+	private NsisDecompressionProvider getDecompressionProvider() throws IOException {
+		if ((this.scriptHeader.compressedHeaderSize & FLAG_IS_COMPRESSED) != 0) { // Check if MSB is
```

Comment:
Imo, the comment about MSB creates more confusion than clarify at this point, as MSB part is hidden by FLAG_IS_COMPRESSED. The code is readable as is, the comment is not needed.

Reply:
I agree, with the name of the bitmask it is pretty clear without the comment